app/services/change_schedule.rb

```diff
@@ -221,16 +226,37 @@ def relevant_induction_record_has_different_schedule
 
   def validate_school_cohort_exists
     return unless participant_profile&.ecf?
-    return if school_default_partnership.present?
+    return if school_partnership.present?
 
     errors.add(:cohort, I18n.t(:missing_school_cohort_default_partnership))
   end
 
-  def school_default_partnership
+  def school_partnership
     return if school.blank?
 
-    school
-      .active_partnerships
-      .find_by(cohort:, relationship: false, lead_provider_id: cpd_lead_provider.lead_provider_id)
+    @school_partnership ||= (school_partnership_without_relationship || school_partnership_including_relationship)
```

Comment:
```suggestion
    @school_partnership ||= school_partnership_without_relationship || school_partnership_including_relationship
```
merging a rubocop update that doesn't allow that